.pre-commit-config.yaml

```diff
@@ -1,15 +1,15 @@
 default_language_version:
-  python: python3.9
+  python: python3.11
```

Comment:
If we are updating dev environments to python 3.11, shouldn't we just target that python version exclusively? I have the feeling it will be easier for us not to make mistakes if we code using the smallest possible version targeted by the package (that way we avoid inadvertently introducing features which are only supported by the later version).

I am fine with this, but in that case we should change the version range in pyproject.toml, etc, and maybe let's confirm one last time with everyone?

+1 CodeRabbit 😆